packages/@uppy/provider-views/src/utils/PartialTreeUtils/getCheckedFilesWithPaths.ts

```diff
@@ -0,0 +1,73 @@
+/* eslint-disable no-param-reassign */
+import type {
+  PartialTree,
+  PartialTreeFile,
+  PartialTreeFolderNode,
+  PartialTreeId,
+} from '@uppy/core/lib/Uppy'
+import type { CompanionFile } from '@uppy/utils/lib/CompanionFile'
+
+export interface Cache {
+  [key: string]: (PartialTreeFile | PartialTreeFolderNode)[]
+}
+
+const getPath = (
+  partialTree: PartialTree,
+  id: PartialTreeId,
+  cache: Cache,
+): (PartialTreeFile | PartialTreeFolderNode)[] => {
+  const sId = id === null ? 'null' : id
+  if (cache[sId]) return cache[sId]
+
+  const file = partialTree.find((f) => f.id === id)!
+
+  if (file.type === 'root') return []
+
+  const meAndParentPath = [...getPath(partialTree, file.parentId, cache), file]
+  cache[sId] = meAndParentPath
+  return meAndParentPath
+}
+
+// See "Uppy file properties" documentation for `.absolutePath` and `.relativePath`
+// (https://uppy.io/docs/uppy/#working-with-uppy-files)
+const getCheckedFilesWithPaths = (
+  partialTree: PartialTree,
+): CompanionFile[] => {
+  const cache: Cache = {}
```

Comment:
```suggestion
  const cache: Cache = {__proto__: null}
```

Reply:
I think `{}` is easier to read?
What situation are you worried about wrt `{}`?

Reply:
If a file has an `id` suh as `getOwnProperty` or `constructor`, we'd get weird behavior. When using null-prototype object, the JS engine can optimize for dictionary mode (at least V8 used to do that)

Reply:
Can confirm, the following test fails with `id=hasOwnProperty`.

<img width="822" alt="image" src="https://github.com/transloadit/uppy/assets/7578559/a07fb47f-e56f-441b-a4ba-fcabb8b7fe46">

<img width="803" alt="image" src="https://github.com/transloadit/uppy/assets/7578559/66fb3867-3c65-4ed1-888b-4da0bbcef1a2">

Sounds good, I'll change it to `Object.create(null)`.

Reply:
Done.